grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -675,22 +675,18 @@ class StockMovementService {
                 'max': max ? max.toInteger() : null,
         ]);
 
-        def editItemsIds = data.collect { it.id }
+        def productsMap = Product.findAllByIdInList(data.collect { it.product_id })
+                .inject([:]) {map, item -> map << [(item.id): item]}
 
-        def substitutionItemsMap = dataService.executeQuery("""
+        def editPageItems = data.collect {
+            def substitutionItems = dataService.executeQuery("""
```

Comment:
you will send n request to the DB instead of one, it will affect the performance

Reply:
Yes, this would be a very bad performance degradation.